src/transform.php

```diff
@@ -28,12 +28,18 @@ function ($carry, $fn) {
  * the result.
  *
  * @param string $methodName
- * @param array  $arguments
+ * @param array  $args
  *
  * @return callable
  */
 function callMethod($methodName, ...$args)
 {
+    if (!is_string($methodName)) {
```

Comment:
Yes, specific Exception would be good. I did not wanted to add new files in this PR that is already messy as is :)